frontend/lib/util/filtering.cpp

```diff
@@ -35,20 +35,19 @@ namespace {
     // We write the regexp so the relevant part of "content" will be captured by
     // std::smatch if we wish to use this to filter out the markup.
 
-    // TODO: Support explicit title and reference targets like in reST direct hyperlinks (and having only target
-    //       show up in sanitized message).
-
-    // There are three "kinds" of content we consider, so we OR these together
+    // There are four "kinds" of content we consider, so we OR these together
     // Capture a normal Chapel identifier
     std::string reCntType1 = R"#(([\w$.]+))#";
     // If it starts with ~, capture last identifier to the right of a '.'
     // Note in a repeated capture group (e.g. `(\w\.?)+` only the last iteration is captured
     std::string reCntType2 = R"#(~([\w$]+\.?)+)#";
     // Starts with !, capture identifier to right of ! without capturing ! itself
     std::string reCntType3 = R"#(!([\w$.]+))#";
+    // support explicit title and reference target, capture the target
+    std::string reCntType4 = R"#((?:[\w$.]+)<([\w$.]+)>)#";
```

Comment:
`reCntType4` is for matching: `title<target>`

I'm not sure, but I'm guessing Sphinx let's the "title" part contain  non word characters like dashes, slashes, and maybe even spaces.

Could you check on that? If that's the case then I think the regexp would be something like this (and if spaces are allowed remove the `\s`):

```
((?:[^<\s]+)<([\w$.]+)>)
```

Reply:
You are right, Sphinx does allow non word characters and spaces, but does not allow spaces before the '<'. I have updated the regex and tests accordingly.